### PR TITLE
Add Book Functionality v1

### LINKED ISSUE #95: Add Books
## Summary

Add functionality to add books. This will require update on the Frontend side, 

## Motivation

It's a core functionality of the CRUD (create, read, update, and delete) operations for persistent storage.

## Describe alternatives you've considered

N/A

### PR DESCRIPTION
**Pull request by PaDiOc18**
### Description of the Change

Add Book Functionality. This is the v1 for the add functionality, we can refactor a lot of code here. Like for example: Error handling, set less variables, etc. But in order to do that we need to sync up with the team.

### Verification Process

1. Go to Manage, and click on `Add`.

ITJLIB1

2. Pop up a Dialog with all info required
ITJLIB2

3. Once you filled all required information, click on `Add`
ITJLIB3

4. It will display book added or something like that,

ITJLIB4

### Applicable Issues

- Closes #95 

### Release Notes

Add books functionality

### CODE REVIEW

File: src/server/api/routers/books.ts [hunk 1]
```diff
@@ -109,12 +128,68 @@ export const booksRouter = createTRPCRouter({
...
+        });
+
+        // If exist then we need to add book on Inventory table only
+        if (exist != null && exist.id != null){
+          input.inventoryData.bookId = exist.id;
+          const responseInventory = await ctx.prisma.inventory.create({
+            data: input.inventoryData
+          });
+          console.log('Book exist, data added to Inventory');
+          // console.log(responseInventory); // Enable for debug
```
[the remark this answers is not in the document]
Comment (PaDiOc18) [reply src/server/api/routers/books.ts:155]: zi

[the remark this answers is not in the document]
Comment (PaDiOc18) [reply src/server/api/routers/books.ts]: Listo

**Review by adlaiCarmona — APPROVED** [review]

File: src/components/dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+                  onChange = {(e) => setTextPublishDates(e)}    
+                  slotProps={{
+                  textField: {
+                    id: 'publishDates',
+                  },
+                  }}/>
+                </LocalizationProvider>
+              </Grid>
+              <Grid item xs={6}>
+                <TextField type="number" id="numberOfPages" label="Number Of Pages" variant="outlined" value = { bookData?.numberOfPages } onChange = {(e) => setBookData({...bookData, numberOfPages: e.target.value}) } fullWidth />
```
Comment (adlaiCarmona) [inline src/components/dialog.tsx:337]: Aquí salió error en vercel, porque e.target.value es string y el type es number
`Type error: Type 'string' is not assignable to type 'number'.`

**Review by adlaiCarmona — CHANGES_REQUESTED** [review]

File: src/components/dialog.tsx (hunk 2 above)
Comment (PaDiOc18) [reply src/components/dialog.tsx:337]: Ah ya,deja lo arreglo

Comment (PaDiOc18) [reply src/components/dialog.tsx:337]: Ya quedo @adlaiCarmona

Comment (PaDiOc18) [reply src/components/dialog.tsx:337]: Deja arreglo el conflicto tambien

**Review by adlaiCarmona — APPROVED** [review]

**Review by NellyQuino — APPROVED** [review]

**Pull request merged**